### ISSUE TITLE
[a11y]: VoiceOver not announcing HeaderMenuItem text on Safari browser

### ISSUE THREAD

**Issue by kenneth-nguyen1** [carbon-design-system/carbon#21007]
### Package

@carbon/react, @carbon/icons-react

### Browser

Safari

### Operating System

MacOS

### Package version

^1.92.1

### React version

18.2.0

### Automated testing tool and ruleset

AVT Audit with manual screen reader testing

### Assistive technology

MacOS VoiceOver

### Description

On Safari browser, MacOS VoiceOver is not announcing the `HeaderMenuItem` text as it receives keyboard focus. In this recording, there is a "Help" `HeaderGlobalAction` that nests a `HeaderMenu` with a few `HeaderMenuItems`. We see that the help button is announced but the menu items are not read after receiving focus.

https://github.com/user-attachments/assets/09934968-994e-45db-a3b4-30e97770a457

---

Tested this interaction on Chrome using MacOS VoiceOver and the menu item text is read

https://github.com/user-attachments/assets/6cdfaec9-d3fa-4b71-b880-d16fa9cf3483

### WCAG 2.1 Violation

WCAG Mapping: 4.1.2 Name, Role, Value

### Reproduction/example

https://stackblitz.com/edit/github-jgsfyz2w?file=package.json

### Steps to reproduce

1. On Safari browser with the stackblitz link, use MacOS VoiceOver to navigate to the "Help" header global action
2. Open the menu dropdown from the "Help" header global action
3. Tab through the menu items and see that VO does not announce the menu item texts

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
